doc/src/snippets/policies.cpp

```diff
@@ -1,93 +1,91 @@
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-
-
-Distributed under the Boost Software License, Version 1.0.
-(See accompanying file Licence.txt or copy at
-http://www.boost.org/LICENSE_1_0.txt)
-*/
-
 #include "../../../include/outcome.hpp"
+#include "outcome.hpp"
 
-#include <iostream>
+namespace outcome = OUTCOME_V2_NAMESPACE;
 
-using namespace OUTCOME_V2_NAMESPACE;
-
-//! [blank]
-struct blank_policy
+//! [abort_policy]
+struct abort_policy : outcome::policy::base
 {
-  //! Performs a narrow check of state, used in the assume_value() functions.
-  template <class Impl> static constexpr void narrow_value_check(Impl &&self) noexcept;
-
-  //! Performs a narrow check of state, used in the assume_error() functions.
-  template <class Impl> static constexpr void narrow_error_check(Impl &&self) noexcept;
-
-  //! Performs a wide check of state, used in the value() functions.
-  template <class Impl> static constexpr void wide_value_check(Impl &&self);
-
-  //! Performs a wide check of state, used in the error() functions.
-  template <class Impl> static constexpr void wide_error_check(Impl &&self);
-
-
-  /******** These only used if the policy is used in Outcome ********/
+  template <class Impl> static constexpr void wide_value_check(const Impl& self)
+  {
+    if (!base::_has_value(self))
+      std::abort();
+  }
 
-  //! Performs a narrow check of state, used in the assume_exception() functions.
-  template <class Impl> static constexpr void narrow_exception_check(Impl &&self) noexcept;
+  template <class Impl> static constexpr void wide_error_check(const Impl& self)
+  {
+    if (!base::_has_error(self))
+      std::abort();
+  }
 
-  //! Performs a wide check of state, used in the exception() functions.
-  template <class Impl> static constexpr void wide_exception_check(Impl &&self);
+  template <class Impl> static constexpr void wide_exception_check(const Impl& self)
+  {
+    if (!base::_has_exception(self))
+      std::abort();
+  }
 };
-//! [blank]
-
-using MyOutcomeType = outcome<int>;
+//! [abort_policy]
 
-//! [cast]
-template <class T, class EC, class E> struct outcome_policy
+//! [throwing_policy]
+template <typename T, typename EC, typename EP>
+struct throwing_policy : outcome::policy::base
 {
-  /*! Performs a wide check of state, used in the value() functions.
+  static_assert(std::is_convertible<EC, std::error_code>::value, "only EC = error_code");
 
-  \effects If outcome does not have a value, if it has an exception it rethrows it via `std::rethrow_exception()`,
-  if has an error it throws a `std::system_error(error())`, else it throws `bad_outcome_access`.
-  */
-  template <class Impl> static constexpr void wide_value_check(const Impl &self)
+  template <class Impl> static constexpr void wide_value_check(const Impl& self)
```

Comment:
I think these won't compile for the rvalue .value() observers etc because you have pinned Impl to const&.

Reply:
But a reference to const can bind to an rvalue.